File: Prefix Sum/Q1. Find the Highest Altitude/largestAltitude.js
/**
 * @param {number[]} gain
 * @return {number}
 */
var largestAltitude = function (gain) {
    //n+1
    const len = gain.length;
    let arr = new Array(len).fill(0)

    for (let i = 0; i < len; i++) {
        arr[i + 1] = arr[i] + gain[i]
    }
    return Math.max(...arr)
    // let max = arr[0]

    // for(let i=1; i<arr.length; i++){
    //     if(arr[i] > max) max = arr[i]
    // }
    // return max
};

File: Prefix Sum/Q1. Find the Highest Altitude/largestAltitude.cpp
using namespace std
#include <iostream>
#include <vector>
#include <algorithm>


class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        const int sz = gain.size();
        vector<int> arr(sz+1,0);

        for(int i=0; i<sz; i++){
            arr[i+1] = arr[i] + gain[i];
        }
        int ans = arr[0];

        for(int i=1; i<sz+1; i++){
            // if(max < arr[i])max = arr[i];
            ans = max(ans, arr[i]);
        }
        return ans;
    }
};
